src/LanguageNegotiationSetter.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Drupal\oe_multilingual;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Language\LanguageInterface;
+
+/**
+ * Helper service for language negotiation method configuration.
+ */
+class LanguageNegotiationSetter {
+
+  /**
+   * Configuration name.
+   */
+  const CONFIG_NAME = 'language.types';
+
+  /**
+   * Drupal\Core\Config\ConfigFactoryInterface definition.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * Constructs a new LanguageNegotiationSetter object.
+   */
+  public function __construct(ConfigFactoryInterface $config_factory) {
+    $this->configFactory = $config_factory;
+  }
+
+  /**
+   * Enable given language negotiation methods.
+   *
+   * @param array $methods
+   *   Array of language negotiation method names.
```

Comment:
Maybe provide an example here of how the passed data should look? By reading this documentation I have no idea of the kind of method names I should be passing in.

Reply:
OK there is ample documentation now, thanks! :+1: